prow/githuboauth/githuboauth.go

```diff
@@ -42,6 +41,24 @@ const (
 	stateKey           = "state"
 )
 
+// Config is a config for requesting users access tokens from GitHub API. It also has
+// a Cookie Store that retains user credentials deriving from GitHub API.
+type Config struct {
+	ClientID     string   `json:"client_id"`
+	ClientSecret string   `json:"client_secret"`
+	RedirectURL  string   `json:"redirect_url"`
+	Scopes       []string `json:"scopes,omitempty"`
+
+	CookieStore *sessions.CookieStore `json:"-"`
+}
+
+// InitGitHubOAuthConfig creates an OAuthClient using GitHubOAuth config and a Cookie Store
+// to retain user credentials.
+func (c *Config) InitGitHubOAuthConfig(cookie *sessions.CookieStore) {
+	gob.Register(&oauth2.Token{})
```

Comment:
I'm unfamiliar with this package and don't understand what's happening here. Can you comment this code with an explanation?

Reply:
The `oauth2.Token` needs to be stored in the CookieStore with a specific encoder. Since we are using `github.com/gorilla/sessions` which uses `github.com/gorilla/securecookie`, it has to be `encoding/gob`.

See https://github.com/gorilla/securecookie/blob/master/doc.go#L56-L59